app/models/asset.rb

```diff
@@ -473,7 +472,7 @@ def unlock
 
   def unlock_expired
     with_lock do
-      if !lock_ttl.nil? && lock_ttl >= Time.now.to_i
+      if !lock_ttl.nil? && lock_ttl <= Time.now.to_i
```

Comment:
Probably only '<' is needed here

Reply:
I believe it is ok as it is, lock should be still valid on `LOCK_DURATION`.

Reply:
Exactly, when you use '<=' and if now == lock_ttl, the lock will be removed, but it should be still valid